report_docx/report/report_docx.py

```diff
@@ -0,0 +1,306 @@
+            tmp_folder_name, output_report[output_type])
+
+    def _generate_doc_reports(
+        self, cr, uid, context, tmp_folder_name,
+        datas, output_type, output_report
+    ):
+        if len(datas) > 1:
+            raise Exception(_(
+                "Multiple docx files export doesn't support. " +
+                "Please export single docx file."))
+        temp_report = self._convert_single_report(
+            cr, uid, context, tmp_folder_name, 0, datas[0], output_type)
+
+        cmd = "mv %s %s" % (
+            tmp_folder_name + temp_report,
+            tmp_folder_name + output_report[output_type])
+
+        os.system(cmd)
+
+    def _combine_pdf_files(self, tmp_folder_name, output_report):
+        output_path = tmp_folder_name + output_report
+        output_temp_path = tmp_folder_name + 'temp.pdf'
+
+        cmd = "gs -q -dNOPAUSE -sDEVICE=pdfwrite -sOUTPUTFILE=%s \
+            -dBATCH %s*water*.pdf" % (output_temp_path, tmp_folder_name)
+        os.system(cmd)
+
+        # remove the last empty page
+        input_stream = PdfFileReader(file(output_temp_path, 'rb'))
+        output_stream = PdfFileWriter()
+
+        pagenum = input_stream.getNumPages()
+        for i in range(pagenum - 1):
+            page = input_stream.getPage(i)
+            output_stream.addPage(page)
+
+        out_stream = file(output_path, 'wb')
+        try:
+            output_stream.write(out_stream)
+        finally:
+            out_stream.close()
+
+    def _convert_single_report(
+        self, cr, uid, context, tmp_folder_name, count, data, output_type
+    ):
+        docx_template_name = 'template_%s.docx' % count
+        convert_docx_file_name = 'convert_%s.docx' % count
+        convert_pdf_file_name = 'convert_%s.pdf' % count
+        pdf_file_with_watermark = 'convert_watermark_%s.pdf' % count
+        watermark_file = 'watermark.pdf'
+
+        self._convert_docx_from_template(
+            cr, uid, data, context,
+            tmp_folder_name,
+            docx_template_name, convert_docx_file_name)
+
+        if output_type == 'pdf':
+            self._convert_docx_to_pdf(
+                tmp_folder_name,
+                convert_docx_file_name
+            )
+
+            self._create_watermark_pdf(
+                cr, uid, context,
+                tmp_folder_name, watermark_file)
+
+            self._add_watermark_to_pdf(
+                tmp_folder_name, watermark_file,
+                convert_pdf_file_name, pdf_file_with_watermark
+            )
+
+            return pdf_file_with_watermark
+
+        return convert_docx_file_name
+
+    def _convert_docx_from_template(
+        self, cr, uid, data, context,
+        tmp_folder_name,
+        docx_template_name, convert_docx_file_name
+    ):
+        action_id = context['params']['action']
+        action = self.pool.get('ir.actions.report.xml').browse(
+            cr, uid, action_id, context)
+
+        template_path = tmp_folder_name + docx_template_name
```

Comment:
File paths should be joined using `os.join` in order to guarantee multi-OS operability.